ultravox/model/ultravox_model.py

```diff
@@ -510,6 +516,7 @@ class ModifiedWhisperEncoder(whisper.WhisperEncoder):
     """
 
     base_model_prefix = "model.encoder"
+    _no_split_modules = ["WhisperEncoderLayer"]
```

Comment:
For my own knowledge, this means Whisper won't be sharded across GPUs (since we don't need it to be?)

Reply:
Actually my understanding is that it will be.
It's just saying where to shard: at the encoder-layer level for Whisper and at the decoder-layer level in Llama. So each layer is on one GPU and you don't split further.

I believe it's also saying the layers that are not part of this (e.g. embedding layers or convs) will not be sharded at all.